src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const cleanedString = sourceString.replace(/;\s*;/g, ';');
+  const regex = /\s*([^:\s]+)\s*:\s*([^;]+)\s*;?/g;
+  let match;
+
+  while ((match = regex.exec(cleanedString)) !== null) {
+    const key = match[1].includes(';', 0)
+      ? match[1].trim().slice(1)
+      : match[1].trim();
+    const value = match[2].includes(';', 0)
+      ? match[2].trim().slice(1)
+      : match[2].trim();
+
+    result[key] = value;
```

Comment:
Using a while loop with regex.exec is not recommended for this task. Instead, consider using string methods like 'split' and 'map' to create an array of declarations and then reduce it to an object.